mcp.go

```diff
@@ -0,0 +1,909 @@
+		// In a real implementation, you would compare with supported versions
+		s.protocolVersion = params.ProtocolVersion
+	}
+
+	// Determine if we have tools and resources
+	hasTools := len(s.toolCmds) > 0
+	hasResources := len(s.resourceCmds) > 0 || len(s.resourceTemplates) > 0
+
+	// Create capabilities object following MCP 2025-03-26 spec
+	capabilities := map[string]interface{}{}
+
+	if hasTools {
+		capabilities["tools"] = map[string]interface{}{
+			"listChanged": false, // We don't support list change notifications yet
+		}
+	}
+
+	if hasResources {
+		capabilities["resources"] = map[string]interface{}{
+			"listChanged": false, // We don't support list change notifications yet
+			"subscribe":   false, // We don't support subscriptions yet
+		}
+	}
+
+	// Create the response
+	result := InitializeResult{
+		ProtocolVersion: s.protocolVersion,
+		ServerInfo: struct {
+			Name    string `json:"name"`
+			Version string `json:"version"`
+		}{
+			Name:    "serpent-mcp",
+			Version: "1.0.0",
+		},
+		Capabilities: capabilities,
+	}
+
+	// Send the response
+	s.sendSuccessResponse(req.ID, result)
+}
+
+// handleInitialized handles the initialized notification
+func (s *MCPServer) handleInitialized(req JSONRPC2Request) {
+	// Mark the server as initialized
+	s.initialized = true
+
+	// No response needed for a notification
+	// But we'll send a success response anyway since our request has an ID
+	if req.ID != nil {
+		s.sendSuccessResponse(req.ID, struct{}{})
+	}
+}
+
+// MCPCommand creates a generic command that can run any serpent command as an MCP server
+func MCPCommand() *Command {
+	return &Command{
+		Use:   "mcp [command]",
+		Short: "Run a command as an MCP server",
+		Long: `Run a command as a Model Context Protocol (MCP) server over stdio.
+
+This command allows any serpent command to be exposed as an MCP server, which can
+provide tools and resources to MCP clients. The server communicates using JSON-RPC 2.0
+over stdin/stdout.
+
+If a command name is provided, that specific command will be run as an MCP server.
+Otherwise, the root command will be used.
+
+Commands with a Tool field set can be invoked as MCP tools.
+Commands with a Resource field set can be accessed as MCP resources.
+Commands with neither Tool nor Resource set will not be accessible via MCP.`,
+		Handler: func(inv *Invocation) error {
+			rootCmd := inv.Command
+			if rootCmd.Parent != nil {
+				// Find the root command
+				for rootCmd.Parent != nil {
+					rootCmd = rootCmd.Parent
+				}
+			}
+
+			// If a command name is provided, use that as the root
+			if len(inv.Args) > 0 {
+				cmdName := strings.Join(inv.Args, " ")
+				cmd := DefaultCommandFinder(rootCmd, cmdName)
+				if cmd == nil {
+					return xerrors.Errorf("command not found: %s", cmdName)
+				}
+				rootCmd = cmd
+			}
+
+			// Create and run the MCP server
+			server := NewMCPServer(rootCmd, inv.Stdin, inv.Stdout, inv.Stderr)
+			return server.Run(inv.Context())
+		},
+	}
+}
```

Comment:
While this is a nice convenience function, isn't this simple enough to just plumb through as required?